Added namespace as a member of Remotable and updated example

JD-ETH opened the pull request:
Added an identifier for remotable class, this allows distinguish instances of the same class/different class sharing the same method as long as user defines a method. 

Alternatively or additionally, we can also add the func name to the identifier to further distinguish between different instantiated classes with the same name. for example PPOAgent.forward() and APPOAgent.forward() wouldn't need an identifier additionally.

xiaomengy commented on examples/tutorials/remote_example.py:
```diff
@@ -59,19 +59,24 @@ async def run_batch(adder_client, send_tensor=False):
 
 def main():
-    adder = Adder()
+    adder1 = Adder('adder1')
```
I'd suggest not to change this file because it is used for most basic tutorials.

xiaomengy commented on rlmeta/core/remote.py:
```diff
@@ -26,15 +26,25 @@ def __new__(cls, name, bases, attrs):
 
 
 class Remotable(abc.ABC, metaclass=RemotableMeta):
+    def __init__(self, identifier=""):
```
Please add type hint for this function.
```
def __init__(self, identifier : str = "") -> None:
```

Also please try to format the file by yapf.

xiaomengy replied:
A high level comment is we have to also modify the __init__ method for the classes derived from Remotable. Or at least add a TODO here.

xiaomengy commented on rlmeta/core/remote.py:
```diff
@@ -26,15 +26,25 @@ def __new__(cls, name, bases, attrs):
 
 class Remotable(abc.ABC, metaclass=RemotableMeta):
+    def __init__(self, identifier=""):
+        self.__identifier__ = identifier
```
I think name it as self._identifier should be enough?

xiaomengy commented:
Link https://github.com/facebookresearch/rlmeta/issues/24

JD-ETH commented:
@xiaomengy  addressed comments. However, I'm unable to run server multiple times in the test that I added, moolib disallows for multiple initialization. 
I'm not sure what's the best solution here. Should i be using server list instead?

xiaomengy commented on tests/core/remotable_test.py:
```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```
Please put all of the test files into tests/

JD-ETH replied:
what do you mean? this is in `tests/core/` already?

xiaomengy commented on rlmeta/core/remote.py:
```diff
@@ -31,10 +31,20 @@ def __new__(cls, name, bases, attrs):
 
 class Remotable(abc.ABC, metaclass=RemotableMeta):
 
+    def __init__(self, identifier: str = ""):
```
RemotableModel is also inherited from Remotable and actually that's the main reason we want this feature.
https://github.com/facebookresearch/rlmeta/blob/main/rlmeta/core/model.py#L17

xiaomengy commented:
> 

Why do you want to run the server for multiple times?

JD-ETH commented:
should we merge this?
